### PR TITLE
Update build definition execute conditions for api v3.2

### PR DESCRIPTION
**Pull request by chcosta**
CoreClr PipeBuild master now runs using VSTS api v3.2. Build definitions need to be updated to use the new api. Some older concepts (like "alwaysRun") are no longer obeyed in the newer api format.
dotnet/core-eng

https://github.com/dotnet/core-eng/issues/2142

### CODE REVIEW

File: buildpipeline/DotNet-CoreClr-Trusted-Linux.json [hunk 1]
```diff
@@ -289,6 +291,7 @@
       "alwaysRun": true,
       "displayName": "Publish Artifact: BuildLogs",
       "timeoutInMinutes": 0,
+      "condition": "succeededOrFailed()",
```
Comment (wtgodbe) [inline buildpipeline/DotNet-CoreClr-Trusted-Linux.json]: What's the difference between `succeededOrFailed()` and `always()` ? Just out of curiosity

Comment (chcosta) [reply buildpipeline/DotNet-CoreClr-Trusted-Linux.json]: succeededorfailed doesn't include cancelled.  always does

**Comment by chcosta** [comment]
@dotnet-bot test 
CentOS7.1 x64 Checked Innerloop Build and Test please

**Comment by chcosta** [comment]
The "CentOS7.1 x64 Checked Innerloop Build and Test" failure is currently failing for all PR's.  My changes do not touch code which is compiled / tested / etc... so I'm merging.

**Pull request merged**